src/datumaro/plugins/data_formats/kaggle/base.py

```diff
@@ -0,0 +1,393 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path as osp
+from typing import Dict, Optional, Type, TypeVar, Union
+
+import numpy as np
+import pandas as pd
+from defusedxml import ElementTree
+
+from datumaro.components.annotation import (
+    AnnotationType,
+    Bbox,
+    Label,
+    LabelCategories,
+    Mask,
+    MaskCategories,
+)
+from datumaro.components.dataset import DatasetItem
+from datumaro.components.dataset_base import DEFAULT_SUBSET_NAME, DatasetBase, SubsetBase
+from datumaro.components.errors import InvalidAnnotationError, InvalidFieldError, MissingFieldError
+from datumaro.components.importer import ImportContext
+from datumaro.components.media import Image, ImageFromFile
+from datumaro.util.image import IMAGE_EXTENSIONS, load_image
+
+T = TypeVar("T")
+
+
+class KaggleImageCsvBase(DatasetBase):
+    def __init__(
+        self,
+        path: str,
+        ann_file: str,
+        columns: Dict[str, str],
+        *,
+        subset: Optional[str] = DEFAULT_SUBSET_NAME,
+        ctx: Optional[ImportContext] = None,
+    ):
+        super().__init__(ctx=ctx)
+
+        self._subset = subset
+
+        self._path = path
+        self._columns = columns
+
+        self._items, label_cat = self._load_items(ann_file, columns)
+        self._categories = {AnnotationType.label: label_cat}
+
+    def _load_items(self, ann_file: str, columns: Dict[str, Union[str, list]]):
+        df = pd.read_csv(ann_file, header=None, on_bad_lines="skip")
+
+        indices = {}
+        for key, field in columns.items():
+            if key == "bbox":
+                indices[key] = []
+                for v in field:
+                    indices[key].append(list(df.iloc[0]).index(v))
+            else:
+                indices[key] = list(df.iloc[0]).index(field)
+
+        label_cat = LabelCategories()
+        items = []
+        for ind, row in df.iterrows():
+            if ind == 0:
+                continue
+
+            data_info = list(row)
+            media_name = data_info[indices["media"]]
+
+            if not media_name.lower().endswith(tuple(IMAGE_EXTENSIONS)):
+                for ext in IMAGE_EXTENSIONS:
+                    if osp.exists(os.path.join(self._path, media_name + ext)):
```

Comment:
Can we maintain this intermediate variable such as
`media_path = os.path.join(self._path, media_name + ext)`
It is being created three times in this code.

Reply:
Will do. Thank you.

Reply:
fixed at 0b13690c5f9a0ad5bcf55e3d6b3baf17ed84c7eb